### PR TITLE
fix: expandables with long titles could overflow chevron

### PR DESCRIPTION
**Pull request by pearofducks**
this branch:

Screenshot 2023-11-13 at 12 35 10

`next` branch:

Screenshot 2023-11-13 at 12 35 02

### CODE REVIEW

File: component-classes/index.js [hunk 1]
```diff
@@ -207,14 +207,14 @@ export const expandable = {
...
-  chevronBox: 'absolute right-16',
+  chevronBox: '',
   chevronTransform: 'transform transition-transform transform-gpu ease-in-out',
   chevronExpand: '-rotate-180',
   chevronCollapse: 'rotate-180',
   expansion: 'overflow-hidden',
   expansionNotExpanded: 'h-0 invisible',
   button: buttonReset + ' hover:underline focus-visible:underline',
-  buttonBox: 'w-full text-left relative inline-flex items-center ' + box.box,
+  buttonBox: 'w-full text-left relative inline-flex items-center space-between ' + box.box,
```
Comment (BalbinaK) [inline component-classes/index.js:217]: I think it should be `justify-between` (see the unocss output snapshot [here](https://github.com/warp-ds/drive/blob/next/test/__snapshots__/position.js.snap#L482))

Comment (pearofducks) [reply component-classes/index.js]: You're totally right! My brains got scrambled copying the fix over from my test env! 🤦

File: component-classes/index.js [hunk 2]
```diff
@@ -207,14 +207,14 @@ export const expandable = {
   expandableBleed: box.bleed,
   chevron: 'inline-block align-middle i-text-$color-expandable-icon',
   chevronNonBox: 'relative left-8',
```
Comment (BalbinaK) [inline component-classes/index.js:209]: Now that we don't apply `absolute` position on chevronBox, `chevronNonBox` can be changed to `ml-8`. This also fixes the following situation in Elements:
Before:
Screenshot 2023-11-13 at 14 48 34

After:
Screenshot 2023-11-13 at 14 50 01

File: component-classes/index.js [hunk 3]
```diff
@@ -207,14 +207,14 @@ export const expandable = {
...
   chevronTransform: 'transform transition-transform transform-gpu ease-in-out',
   chevronExpand: '-rotate-180',
   chevronCollapse: 'rotate-180',
   expansion: 'overflow-hidden',
   expansionNotExpanded: 'h-0 invisible',
   button: buttonReset + ' hover:underline focus-visible:underline',
-  buttonBox: 'w-full text-left relative inline-flex items-center ' + box.box,
+  buttonBox: 'w-full text-left relative inline-flex items-center justify-between ' + box.box,
   paddingTop: 'pt-0',
   title: 'flex justify-between items-center',
```
Comment (BalbinaK) [inline component-classes/index.js:219]: In React and Elements we are wrapping text and chevron in a div, which then holds `expandable.title` classes. In order for the chevron to be positioned at the end of a box Expandable there, we'll need to apply `w-full` here.

**Review by BalbinaK — COMMENTED** [review]
This may work in Vue but will break React and Elements, so here are some comments that should fix it as far as I tested :)

**Comment by siljec** [comment]
Hi @pearofducks! Thanks for fixing this 🙏  We need to focus on some other tickets before we have time to test this properly. Just want to let you know! But we will take a look at it when we have time.

**Comment by pearofducks** [comment]
That's totally fine! Feel free to run with the changes Balbina suggested - I just wanted to get a fix started instead of just reporting a bug - but I'll leave this to the experts to finalize! 🙈

File: component-classes/index.js (hunk 2 above)
Comment (BalbinaK) [reply component-classes/index.js]: [f690a2d](https://github.com/warp-ds/css/pull/89/commits/f690a2d864699570607a2f69abb40839aa1976c8)

File: component-classes/index.js (hunk 3 above)
Comment (BalbinaK) [reply component-classes/index.js]: [08e6137](https://github.com/warp-ds/css/pull/89/commits/08e613745b78b2b8ad64c6d0820316afb52f152a)

**Comment by BalbinaK** [comment]
I had the changes I had tested on my local repo anyway, so I double-checked that everything still works and pushed. This PR should be good to merge:
Vue
Screenshot 2023-11-15 at 13 40 10
React
Screenshot 2023-11-15 at 13 40 54
Elements
Screenshot 2023-11-15 at 13 41 42

**Comment by pearofducks** [comment]
Awesome job @BalbinaK ! I'll let y'all merge this whenever suits, thanks so much for help resolving!

**Review by imprashast — APPROVED** [review]

**Pull request merged**